### PR TITLE
(PC-14140) feat(booking): new display no ticket collect

### PR DESCRIPTION
**Pull request by cedricls1**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-14140

## Checklist

I have:

- [ ] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** or **github dev name** for any added TODO / FIXME.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)
- [ ] Made sure translations file is up to date (`yarn translations:extract`)

## Screenshots

Capture d’écran 2022-04-11 à 16 52 32

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/ui/svg/icons/BicolorCircledCheck.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import * as React from 'react'
+import Svg, { Defs, LinearGradient, Stop, Path } from 'react-native-svg'
+import styled from 'styled-components/native'
+
+import { svgIdentifier } from 'ui/svg/utils'
+
+import { BicolorIconInterface } from './types'
+
+const BicolorCircledCheckSvg: React.FC<BicolorIconInterface> = ({
```
[the remark this answers is not in the document]
Comment (cedricls1) [reply src/ui/svg/icons/BicolorCircledCheck.tsx:9]: Elle a bien été ajoutée (c'est une illustration)

File: src/features/bookings/atoms/TicketCode.tsx [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
 
-export function TicketCode({ code }: { code: string }) {
-  return <TicketCodeTitle>{code}</TicketCodeTitle>
+type TicketCodeProps = {
+  code: string
+  collectType: string
+}
+
+export function TicketCode({ code, collectType }: TicketCodeProps) {
+  if (collectType === '' || collectType === WithdrawalTypeEnum.on_site) {
```
[the remark this answers is not in the document]
Comment (cedricls1) [reply src/features/bookings/atoms/TicketCode.tsx:12]: Dans l'API on a WithdrawalTypeEnum | null | undefined et en gros je vais afficher le code si on n'a pas de withdrawal type ou si on est sur un withdrawal type retrait sur place

File: src/features/bookings/components/BookingDetailsTicketContent.tsx [hunk 3]
```diff
@@ -49,7 +50,7 @@ export const BookingDetailsTicketContent = (props: BookingDetailsTicketContentPr
   const isDigitalAndActivationCodeEnabled =
     activationCodeFeatureEnabled && properties.hasActivationCode
 
-  const collectType = booking?.stock?.offer?.withdrawalType || ''
+  const collectType = booking?.stock?.offer?.withdrawalType || null
```
Comment (bruno-ebstein-pass-culture) [inline src/features/bookings/components/BookingDetailsTicketContent.tsx:53]: je pense qu'`undefined` c'est une bonne valeur par défaut

```suggestion
  const collectType = booking?.stock?.offer?.withdrawalType
```

Comment (bruno-ebstein-pass-culture) [reply src/features/bookings/components/BookingDetailsTicketContent.tsx:53]:
```suggestion
  const collectType = booking?.stock?.offer?.withdrawalType || undefined
```

**Review by bruno-ebstein-pass-culture — APPROVED** [review]

**Pull request merged**